src/driver/sqljs/SqljsDriver.ts

```diff
@@ -0,0 +1,226 @@
+                return this.createDatabaseConnectionWithImport(JSON.parse(localStorageContent));
+            }
+        }
+        else {
+            return this.createDatabaseConnectionWithImport(fileNameOrLocalStorageOrData);
+        }
+    }
+
+    /**
+     * Saved the current database to the given file (Node.js) or local storage key (browser).
+     * If no location path is given, the location path in the options (if specified) will be used.
+     */
+    async save(location?: string) {
+        if (!location && !this.options.location) {
+            throw new Error(`No location is set, specify a location parameter or add the location option to your configuration`);
+        }
+        
+        let path = "";
+        if (location) {
+            path = location;
+        }
+        else if (this.options.location) {
+            path = this.options.location;
+        }
+
+        if (PlatformTools.type === "node") {
+            try {
+                const content = new Buffer(this.databaseConnection.export());
+                await PlatformTools.writeFile(path, content);
+            }
+            catch (e) {
+                throw new Error(`Could not save database, error: ${e}`);
+            }
+        }
+        else {
+            const database: Uint8Array = this.databaseConnection.export();
+            // convert Uint8Array to number array to improve local-storage storage
+            const databaseArray = [].slice.call(database);
+            PlatformTools.getGlobalVariable().localStorage.setItem(path, JSON.stringify(databaseArray));
+        }
+    }
+
+    /**
+     * This gets called by the QueryRunner when a change to the database is made.
+     * If a custom autoSaveCallback is specified, it get's called with the database as Uint8Array,
+     * otherwise the save method is called which saves it to file (Node.js) or localstorage (browser).
+     */
+    async autoSave() {
+        if (this.options.autoSave) {
+            if (this.options.autoSaveCallback) {
+                await this.options.autoSaveCallback(this.export());
+            }
+            else {
+                await this.save();
+            }
+        }
+    }
+    
+    /**
+     * Returns the current database as Uint8Array.
+     */
+    export(): Uint8Array {
+        return this.databaseConnection.export();
+    }
+
+    /**
+     * Creates generated map of values generated or returned by database after INSERT query.
+     */
+    createGeneratedMap(metadata: EntityMetadata, insertResult: any) {
+        const generatedMap = metadata.generatedColumns.reduce((map, generatedColumn) => {
+            // seems to be the only way to get the inserted id, see https://github.com/kripken/sql.js/issues/77
+            if (generatedColumn.isPrimary && generatedColumn.generationStrategy === "increment") {
+                const query = "SELECT last_insert_rowid()";
+                let result = this.databaseConnection.exec(query);
```

Comment:
Query Execution is synchronous?...

Reply:
Sadly yes. I have to fake the asynchronous behavior with this driver